src/main/java/org/gradle/profiler/jfr/JfrProfiler.java

```diff
@@ -45,4 +46,12 @@ protected JvmArgsCalculator jvmArgsWithInstrumentation(ScenarioSettings settings
     private File getJfrFile(ScenarioSettings settings) {
         return new File(settings.getScenario().getOutputDir(), settings.getScenario().getProfileName() + PROFILE_JFR_SUFFIX);
     }
+
+    @Override
+    public void validate(ScenarioSettings settings, Consumer<String> reporter) {
+        GradleScenarioDefinition scenario = settings.getScenario();
```

Comment:
Could you extract this validation into a method, so you don't need to copy all the code here?